packages/core/src/scp-cf/destination-accessor.ts

```diff
@@ -482,19 +285,346 @@ const emptyDestinationByType: DestinationsByType = {
   subaccount: []
 };
 
-function subscriberDestinationIsSelected(
-  selectionStrategy: DestinationSelectionStrategy
-): boolean {
-  return (
-    selectionStrategy === subscriberFirst ||
-    selectionStrategy === alwaysSubscriber
-  );
-}
-
 function emptyDestinationsByType(
   destinationByType: DestinationsByType
 ): boolean {
   return (
-    !destinationByType.instance.length && !destinationByType.instance.length
+    !destinationByType.instance.length && !destinationByType.subaccount.length
   );
 }
+
+class DestinationAccessor {
+  public static async getDestinationFromDestinationService(
+    name: string,
+    options: DestinationOptions & { iss?: string }
+  ): Promise<Destination | undefined> {
+    const decodedUserJwt = await DestinationAccessor.getDecodedUserJwt(options);
+    const providerToken = await DestinationAccessor.getProviderToken(options);
+    const da = new DestinationAccessor(
+      name,
+      options,
+      decodedUserJwt,
+      providerToken
+    );
+
+    return (
+      da.trySubscriberCache() ??
+      (await da.tryProviderCache()) ??
+      (await da.processDestinationResult(await da.tryForOAuth())) ??
+      (await da.processDestinationResult(await da.tryForCert())) ??
+      (await da.processDestinationResult(await da.tryService())) ??
+      undefined
+    );
```

Comment:
`tryForOAuth` and `tryForCert` both call `tryService`. Also, `processDestinationResult` is always called.
```suggestion
    if (this.options.useCache) {
      return da.trySubscriberCache() || (await da.tryProviderCache());
      }
      let destination = await da.tryService();
      destination = await da.tryForOAuth(destination) || await da.tryForCert(destination) || destination;
      return da.processDestinationResult(destination);
    );
```

Reply:
I tried to make it more clear